udebs/addons/openvpn/source/openvpn.8
.\"  OpenVPN -- An application to securely tunnel IP networks
.\"             over a single TCP/UDP port, with support for SSL/TLS-based
.\"             session authentication and key exchange,
.\"             packet encryption, packet authentication, and
.\"             packet compression.
.\"
.\"  Copyright (C) 2002-2005 OpenVPN Solutions LLC <info@openvpn.net>
.\"
.\"  This program is free software; you can redistribute it and/or modify
.\"  it under the terms of the GNU General Public License version 2
.\"  as published by the Free Software Foundation.
.\"
.\"  This program is distributed in the hope that it will be useful,
.\"  but WITHOUT ANY WARRANTY; without even the implied warranty of
.\"  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\"  GNU General Public License for more details.
.\"
.\"  You should have received a copy of the GNU General Public License
.\"  along with this program (see the file COPYING included with this
.\"  distribution); if not, write to the Free Software Foundation, Inc.,
.\"  59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
.\"
.\" Manual page for openvpn
.\" SH section heading
.\" SS subsection heading
.\" LP paragraph
.\" IP indented paragraph
.\" TP hanging label
.TH openvpn 8 "3 August 2005"
.\"*********************************************************
.SH NAME
openvpn \- secure IP tunnel daemon.
.\"*********************************************************
.SH SYNOPSIS
.LP
.nh
.in +4
.ti -4
.B openvpn
[\ \fB\-\-help\fR\ ]
.in -4
.ti +4
.hy

.nh
.in +4
.ti -4
.B openvpn
[\ \fB\-\-config\fR\ \fIfile\fR\ ]
.in -4
.ti +4
.hy

.nh
.in +4
.ti -4
.B openvpn
[\ \fB\-\-genkey\fR\ ]
[\ \fB\-\-secret\fR\ \fIfile\fR\ ]
.in -4
.ti +4
.hy

.nh
.in +4
.ti -4
.B openvpn
[\ \fB\-\-mktun\fR\ ]
[\ \fB\-\-rmtun\fR\ ]
[\ \fB\-\-dev\fR\ \fItunX\ |\ tapX\fR\ ]
[\ \fB\-\-dev\-type\fR\ \fIdevice\-type\fR\ ]
[\ \fB\-\-dev\-node\fR\ \fInode\fR\ ]
.in -4
.ti +4
.hy

.nh
.in +4
.ti -4
.B openvpn
[\ \fB\-\-test\-crypto\fR\ ]
[\ \fB\-\-secret\fR\ \fIfile\fR\ ]
[\ \fB\-\-auth\fR\ \fIalg\fR\ ]
[\ \fB\-\-cipher\fR\ \fIalg\fR\ ]
[\ \fB\-\-engine\fR\ ]
[\ \fB\-\-keysize\fR\ \fIn\fR\ ]
[\ \fB\-\-no\-replay\fR\ ]
[\ \fB\-\-no\-iv\fR\ ]
.in -4
.ti +4
.hy

.nh
.in +4
.ti -4
.B openvpn
[\ \fB\-\-askpass\fR\ \fI[file]\fR\ ]
[\ \fB\-\-auth\-nocache\fR\ ]
[\ \fB\-\-auth\-retry\fR\ \fItype\fR\ ]
[\ \fB\-\-auth\-user\-pass\-verify\fR\ \fIscript\fR\ ]
[\ \fB\-\-auth\-user\-pass\fR\ \fIup\fR\ ]
[\ \fB\-\-auth\fR\ \fIalg\fR\ ]
[\ \fB\-\-bcast\-buffers\fR\ \fIn\fR\ ]
[\ \fB\-\-ca\fR\ \fIfile\fR\ ]
[\ \fB\-\-ccd\-exclusive\fR\ ]
[\ \fB\-\-cd\fR\ \fIdir\fR\ ]
[\ \fB\-\-cert\fR\ \fIfile\fR\ ]
[\ \fB\-\-chroot\fR\ \fIdir\fR\ ]
[\ \fB\-\-cipher\fR\ \fIalg\fR\ ]
[\ \fB\-\-client\-cert\-not\-required\fR\ ]
[\ \fB\-\-client\-config\-dir\fR\ \fIdir\fR\ ]
[\ \fB\-\-client\-connect\fR\ \fIscript\fR\ ]
[\ \fB\-\-client\-disconnect\fR\ ]
[\ \fB\-\-client\-to\-client\fR\ ]
[\ \fB\-\-client\fR\ ]
[\ \fB\-\-comp\-lzo\fR\ ]
[\ \fB\-\-comp\-noadapt\fR\ ]
[\ \fB\-\-config\fR\ \fIfile\fR\ ]
[\ \fB\-\-connect\-freq\fR\ \fIn\ sec\fR\ ]
[\ \fB\-\-connect\-retry\fR\ \fIn\fR\ ]
[\ \fB\-\-crl\-verify\fR\ \fIcrl\fR\ ]
[\ \fB\-\-cryptoapicert\fR\ \fIselect\-string\fR\ ]
[\ \fB\-\-daemon\fR\ \fI[progname]\fR\ ]
[\ \fB\-\-dev\-node\fR\ \fInode\fR\ ]
[\ \fB\-\-dev\-type\fR\ \fIdevice\-type\fR\ ]
[\ \fB\-\-dev\fR\ \fItunX\ |\ tapX\ |\ null\fR\ ]
[\ \fB\-\-dev\fR\ \fItunX\ |\ tapX\fR\ ]
[\ \fB\-\-dh\fR\ \fIfile\fR\ ]
[\ \fB\-\-dhcp\-option\fR\ \fItype\ [parm]\fR\ ]
[\ \fB\-\-dhcp\-release\fR\ ]
[\ \fB\-\-dhcp\-renew\fR\ ]
[\ \fB\-\-disable\-occ\fR\ ]
[\ \fB\-\-disable\fR\ ]
[\ \fB\-\-down\-pre\fR\ ]
[\ \fB\-\-down\fR\ \fIcmd\fR\ ]
[\ \fB\-\-duplicate\-cn\fR\ ]
[\ \fB\-\-echo\fR\ \fI[parms...]\fR\ ]
[\ \fB\-\-engine\fR\ \fI[engine\-name]\fR\ ]
[\ \fB\-\-explicit\-exit\-notify\fR\ \fI[n]\fR\ ]
[\ \fB\-\-fast\-io\fR\ ]
[\ \fB\-\-float\fR\ ]
[\ \fB\-\-fragment\fR\ \fImax\fR\ ]
[\ \fB\-\-genkey\fR\ ]
[\ \fB\-\-group\fR\ \fIgroup\fR\ ]
[\ \fB\-\-hand\-window\fR\ \fIn\fR\ ]
[\ \fB\-\-hash\-size\fR\ \fIr\ v\fR\ ]
[\ \fB\-\-help\fR\ ]
[\ \fB\-\-http\-proxy\-option\fR\ \fItype\ [parm]\fR\ ]
[\ \fB\-\-http\-proxy\-retry\fR\ ]
[\ \fB\-\-http\-proxy\-timeout\fR\ \fIn\fR\ ]
[\ \fB\-\-http\-proxy\fR\ \fIserver\ port\ [authfile]\ [auth\-method]\fR\ ]
[\ \fB\-\-ifconfig\-noexec\fR\ ]
[\ \fB\-\-ifconfig\-nowarn\fR\ ]
[\ \fB\-\-ifconfig\-pool\-linear\fR\ ]
[\ \fB\-\-ifconfig\-pool\-persist\fR\ \fIfile\ [seconds]\fR\ ]
[\ \fB\-\-ifconfig\-pool\fR\ \fIstart\-IP\ end\-IP\ [netmask]\fR\ ]
[\ \fB\-\-ifconfig\-push\fR\ \fIlocal\ remote\-netmask\fR\ ]
[\ \fB\-\-ifconfig\fR\ \fIl\ rn\fR\ ]
[\ \fB\-\-inactive\fR\ \fIn\fR\ ]
[\ \fB\-\-inetd\fR\ \fI[wait|nowait]\ [progname]\fR\ ]
[\ \fB\-\-ip\-win32\fR\ \fImethod\fR\ ]
[\ \fB\-\-ipchange\fR\ \fIcmd\fR\ ]
[\ \fB\-\-iroute\fR\ \fInetwork\ [netmask]\fR\ ]
[\ \fB\-\-keepalive\fR\ \fIn\ m\fR\ ]
[\ \fB\-\-key\-method\fR\ \fIm\fR\ ]
[\ \fB\-\-key\fR\ \fIfile\fR\ ]
[\ \fB\-\-keysize\fR\ \fIn\fR\ ]
[\ \fB\-\-learn\-address\fR\ \fIcmd\fR\ ]
[\ \fB\-\-link\-mtu\fR\ \fIn\fR\ ]
[\ \fB\-\-local\fR\ \fIhost\fR\ ]
[\ \fB\-\-log\-append\fR\ \fIfile\fR\ ]
[\ \fB\-\-log\fR\ \fIfile\fR\ ]
[\ \fB\-\-suppress-timestamps\fR\ ]
[\ \fB\-\-lport\fR\ \fIport\fR\ ]
[\ \fB\-\-management\-hold\fR\ ]
[\ \fB\-\-management\-log\-cache\fR\ \fIn\fR\ ]
[\ \fB\-\-management\-query\-passwords\fR\ ]
[\ \fB\-\-management\fR\ \fIIP\ port\ [pw\-file]\fR\ ]
[\ \fB\-\-max\-clients\fR\ \fIn\fR\ ]
[\ \fB\-\-max\-routes\-per\-client\fR\ \fIn\fR\ ]
[\ \fB\-\-mktun\fR\ ]
[\ \fB\-\-mlock\fR\ ]
[\ \fB\-\-mode\fR\ \fIm\fR\ ]
[\ \fB\-\-mssfix\fR\ \fImax\fR\ ]
[\ \fB\-\-mtu\-disc\fR\ \fItype\fR\ ]
[\ \fB\-\-mtu\-test\fR\ ]
[\ \fB\-\-mute\-replay\-warnings\fR\ ]
[\ \fB\-\-mute\fR\ \fIn\fR\ ]
[\ \fB\-\-nice\fR\ \fIn\fR\ ]
[\ \fB\-\-no\-iv\fR\ ]
[\ \fB\-\-no\-replay\fR\ ]
[\ \fB\-\-nobind\fR\ ]
[\ \fB\-\-ns\-cert\-type\fR\ \fIclient|server\fR\ ]
[\ \fB\-\-passtos\fR\ ]
[\ \fB\-\-pause\-exit\fR\ ]
[\ \fB\-\-persist\-key\fR\ ]
[\ \fB\-\-persist\-local\-ip\fR\ ]
[\ \fB\-\-persist\-remote\-ip\fR\ ]
[\ \fB\-\-persist\-tun\fR\ ]
[\ \fB\-\-ping\-exit\fR\ \fIn\fR\ ]
[\ \fB\-\-ping\-restart\fR\ \fIn\fR\ ]
[\ \fB\-\-ping\-timer\-rem\fR\ ]
[\ \fB\-\-ping\fR\ \fIn\fR\ ]
[\ \fB\-\-pkcs12\fR\ \fIfile\fR\ ]
[\ \fB\-\-plugin\fR\ \fImodule\-pathname\ init\-string\fR\ ]
[\ \fB\-\-port\fR\ \fIport\fR\ ]
[\ \fB\-\-proto\fR\ \fIp\fR\ ]
[\ \fB\-\-pull\fR\ ]
[\ \fB\-\-push\-reset\fR\ ]
[\ \fB\-\-push\fR\ \fI"option"\fR\ ]
[\ \fB\-\-rcvbuf\fR\ \fIsize\fR\ ]
[\ \fB\-\-redirect\-gateway\fR\ \fI["local"]\ ["def1"]\fR\ ]
[\ \fB\-\-remap\-usr1\fR\ \fIsignal\fR\ ]
[\ \fB\-\-remote\-random\fR\ ]
[\ \fB\-\-remote\fR\ \fIhost\ [port]\fR\ ]
[\ \fB\-\-reneg\-bytes\fR\ \fIn\fR\ ]
[\ \fB\-\-reneg\-pkts\fR\ \fIn\fR\ ]
[\ \fB\-\-reneg\-sec\fR\ \fIn\fR\ ]
[\ \fB\-\-replay\-persist\fR\ \fIfile\fR\ ]
[\ \fB\-\-replay\-window\fR\ \fIn\ [t]\fR\ ]
[\ \fB\-\-resolv\-retry\fR\ \fIn\fR\ ]
[\ \fB\-\-rmtun\fR\ ]
[\ \fB\-\-route\-delay\fR\ \fI[n]\ [w]\fR\ ]
[\ \fB\-\-route\-gateway\fR\ \fIgw\fR\ ]
[\ \fB\-\-route\-method\fR\ \fIm\fR\ ]
[\ \fB\-\-route\-noexec\fR\ ]
[\ \fB\-\-route\-up\fR\ \fIcmd\fR\ ]
[\ \fB\-\-route\fR\ \fInetwork\ [netmask]\ [gateway]\ [metric]\fR\ ]
[\ \fB\-\-rport\fR\ \fIport\fR\ ]
[\ \fB\-\-secret\fR\ \fIfile\ [direction]\fR\ ]
[\ \fB\-\-secret\fR\ \fIfile\fR\ ]
[\ \fB\-\-server\-bridge\fR\ \fIgateway\ netmask\ pool\-start\-IP\ pool\-end\-IP\fR\ ]
[\ \fB\-\-server\fR\ \fInetwork\ netmask\fR\ ]
[\ \fB\-\-service\fR\ \fIexit\-event\ [0|1]\fR\ ]
[\ \fB\-\-setenv\fR\ \fIname\ value\fR\ ]
[\ \fB\-\-shaper\fR\ \fIn\fR\ ]
[\ \fB\-\-show\-adapters\fR\ ]
[\ \fB\-\-show\-ciphers\fR\ ]
[\ \fB\-\-show\-digests\fR\ ]
[\ \fB\-\-show\-engines\fR\ ]
[\ \fB\-\-show\-net\-up\fR\ ]
[\ \fB\-\-show\-net\fR\ ]
[\ \fB\-\-show\-tls\fR\ ]
[\ \fB\-\-show\-valid\-subnets\fR\ ]
[\ \fB\-\-single\-session\fR\ ]
[\ \fB\-\-sndbuf\fR\ \fIsize\fR\ ]
[\ \fB\-\-socks\-proxy\-retry\fR\ ]
[\ \fB\-\-socks\-proxy\fR\ \fIserver\ [port]\fR\ ]
[\ \fB\-\-status\fR\ \fIfile\ [n]\fR\ ]
[\ \fB\-\-status\-version\fR\ \fIn\fR\ ]
[\ \fB\-\-syslog\fR\ \fI[progname]\fR\ ]
[\ \fB\-\-tap\-sleep\fR\ \fIn\fR\ ]
[\ \fB\-\-tcp\-queue\-limit\fR\ \fIn\fR\ ]
[\ \fB\-\-test\-crypto\fR\ ]
[\ \fB\-\-tls\-auth\fR\ \fIfile\ [direction]\fR\ ]
[\ \fB\-\-tls\-cipher\fR\ \fIl\fR\ ]
[\ \fB\-\-tls\-client\fR\ ]
[\ \fB\-\-tls\-exit\fR\ ]
[\ \fB\-\-tls\-remote\fR\ \fIx509name\fR\ ]
[\ \fB\-\-tls\-server\fR\ ]
[\ \fB\-\-tls\-timeout\fR\ \fIn\fR\ ]
[\ \fB\-\-tls\-verify\fR\ \fIcmd\fR\ ]
[\ \fB\-\-tmp\-dir\fR\ \fIdir\fR\ ]
[\ \fB\-\-tran\-window\fR\ \fIn\fR\ ]
[\ \fB\-\-tun\-ipv6\fR\ ]
[\ \fB\-\-tun\-mtu\-extra\fR\ \fIn\fR\ ]
[\ \fB\-\-tun\-mtu\fR\ \fIn\fR\ ]
[\ \fB\-\-txqueuelen\fR\ \fIn\fR\ ]
[\ \fB\-\-up\-delay\fR\ ]
[\ \fB\-\-up\-restart\fR\ ]
[\ \fB\-\-up\fR\ \fIcmd\fR\ ]
[\ \fB\-\-user\fR\ \fIuser\fR\ ]
[\ \fB\-\-username\-as\-common\-name\fR\ ]
[\ \fB\-\-verb\fR\ \fIn\fR\ ]
[\ \fB\-\-writepid\fR\ \fIfile\fR\ ]
.in -4
.ti +4
.hy
.\"*********************************************************
.SH INTRODUCTION
.LP
OpenVPN is an open source VPN daemon by James Yonan.
Because OpenVPN tries to
be a universal VPN tool offering a great deal of flexibility,
there are a lot of options on this manual page.
If you're new to OpenVPN, you might want to skip ahead to the
examples section where you will see how to construct simple
VPNs on the command line without even needing a configuration file.

Also note that there's more documentation and examples on
the OpenVPN web site:
.I http://openvpn.net/

And if you would like to see a shorter version of this manual,
see the openvpn usage message which can be obtained by
running
.B openvpn
without any parameters.
.\"*********************************************************
.SH DESCRIPTION
.LP
OpenVPN is a robust and highly flexible VPN daemon.
OpenVPN supports SSL/TLS security, ethernet bridging,
TCP or UDP tunnel transport through proxies or NAT,
support for dynamic IP addresses and DHCP,
scalability to hundreds or thousands of users,
and portability to most major OS platforms.

OpenVPN is tightly bound to the OpenSSL library, and derives much
of its crypto capabilities from it.

OpenVPN supports
conventional encryption
using a pre-shared secret key
.B (Static Key mode)
or
public key security
.B (SSL/TLS mode)
using client & server certificates.
OpenVPN also
supports non-encrypted TCP/UDP tunnels.  

OpenVPN is designed to work with the
.B TUN/TAP
virtual networking interface that exists on most platforms.

Overall, OpenVPN aims to offer many of the key features of IPSec but
with a relatively lightweight footprint.
.\"*********************************************************
.SH OPTIONS
OpenVPN allows any option to be placed either on the command line
or in a configuration file.  Though all command line options are preceded
by a double-leading-dash ("\-\-"), this prefix can be removed when
an option is placed in a configuration file.
.\"*********************************************************
.TP
.B \-\-help
Show options.
.\"*********************************************************
.TP
.B \-\-config file
Load additional config options from
.B file
where each line corresponds to one command line option,
but with the leading '\-\-' removed.

If
.B \-\-config file
is the only option to the openvpn command,
the
.B \-\-config
can be removed, and the command can be given as
.B openvpn file

Note that
configuration files can be nested to a reasonable depth.

Double quotation characters ("") can be used
to enclose single parameters containing whitespace,
and "#" or ";" characters in the first column
can be used to denote comments.

Note that OpenVPN 2.0 and higher performs backslash-based shell
escaping, so the following mappings should be observed:

.RS
.ft 3
.nf
.sp
\\\\       Maps to a single backslash character (\\).
\\"       Pass a literal doublequote character ("), don't
         interpret it as enclosing a parameter.
\\[SPACE] Pass a literal space or tab character, don't
         interpret it as a parameter delimiter.
.ft
.LP
.RE
.fi

For example on Windows, use double backslashes to
represent pathnames:

.RS
.ft 3
.nf
.sp
secret "c:\\\\OpenVPN\\\\secret.key"
.ft
.LP
.RE
.fi

For examples of configuration files,
see
.I http://openvpn.net/examples.html

Here is an example configuration file:
.RS
.ft 3
.nf
.sp
#
# Sample OpenVPN configuration file for
# using a pre-shared static key.
#
# '#' or ';' may be used to delimit comments.

# Use a dynamic tun device.
dev tun

# Our remote peer
remote mypeer.mydomain

# 10.1.0.1 is our local VPN endpoint
# 10.1.0.2 is our remote VPN endpoint
ifconfig 10.1.0.1 10.1.0.2

# Our pre-shared static key
secret static.key
.ft
.LP
.RE
.fi
.\"*********************************************************
.SS Tunnel Options:
.TP
.B \-\-mode m
Set OpenVPN major mode.  By default, OpenVPN runs in
point-to-point mode ("p2p").  OpenVPN 2.0 introduces
a new mode ("server") which implements a multi-client
server capability.
.\"*********************************************************
.TP
.B \-\-local host
Local host name or IP address.
If specified, OpenVPN will bind to this address only.
If unspecified, OpenVPN will bind to all interfaces.
.\"*********************************************************
.TP
.B \-\-remote host [port]
Remote host name or IP address.  On the client, multiple
.B \-\-remote
options may be specified for redundancy, each referring
to a different OpenVPN server.

The OpenVPN client will try to connect to a server at
.B host:port
in the order specified by the list of
.B \-\-remote
options.

The client will move on to the next host in the list,
in the event of connection failure.
Note that at any given time, the OpenVPN client
will at most be connected to
one server.

Note that since UDP is connectionless, connection failure
is defined by the
.B \-\-ping
and
.B \-\-ping-restart
options.

Note the following corner case:  If you use multiple
.B \-\-remote
options, AND you are dropping root privileges on
the client with
.B \-\-user
and/or
.B \-\-group,
AND the client is running a non-Windows OS, if the client needs
to switch to a different server, and that server pushes
back different TUN/TAP or route settings, the client may lack
the necessary privileges to close and reopen the TUN/TAP interface.
This could cause the client to exit with a fatal error.

If
.B \-\-remote
is unspecified, OpenVPN will listen
for packets from any IP address, but will not act on those packets unless
they pass all authentication tests.  This requirement for authentication
is binding on all potential peers, even those from known and supposedly
trusted IP addresses (it is very easy to forge a source IP address on
a UDP packet).

When used in TCP mode, 
.B \-\-remote
will act as a filter, rejecting connections from any host which does
not match
.B host.

If
.B host
is a DNS name which resolves to multiple IP addresses,
one will be randomly
chosen, providing a sort of basic load-balancing and
failover capability.
.\"*********************************************************
.TP
.B \-\-remote-random
When multiple
.B \-\-remote
address/ports are specified, initially randomize the order of the list
as a kind of basic load-balancing measure.
.\"*********************************************************
.TP
.B \-\-proto p
Use protocol
.B p
for communicating with remote host.
.B p
can be
.B udp,
.B tcp-client,
or
.B tcp-server.

The default protocol is
.B udp
when
.B \-\-proto
is not specified.

For UDP operation,
.B \-\-proto udp
should be specified on both peers.

For TCP operation, one peer must use
.B \-\-proto tcp-server
and the other must use
.B \-\-proto tcp-client.
A peer started with
.B tcp-server
will wait indefinitely for an incoming connection.  A peer
started with
.B tcp-client
will attempt to connect, and if that fails, will sleep for 5
seconds (adjustable via the
.B \-\-connect-retry
option) and try again.  Both TCP client and server will simulate

debian/changelog
backharddi-ng-kernel (0.15) lucid; urgency=low

  * Actualización de backharddi-ng-udeb.
  * Actualización de partman-base.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Wed, 22 Sep 2010 10:03:58 +0200

backharddi-ng-kernel (0.14) lucid; urgency=low

  * Arreglada la visualización de la interfaz gráfica.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Thu, 16 Sep 2010 11:10:07 +0200

backharddi-ng-kernel (0.13) lucid; urgency=low

  * Actualización de backharddi-ng-udeb.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Wed, 01 Sep 2010 10:16:21 +0200

backharddi-ng-kernel (0.12) lucid; urgency=low

  * Actualización de backharddi-ng-udeb.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Tue, 27 Jul 2010 19:55:02 +0200

backharddi-ng-kernel (0.11) lucid; urgency=low

  * Actualización de backharddi-ng-udeb.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Fri, 23 Jul 2010 10:38:02 +0200

backharddi-ng-kernel (0.10) lucid; urgency=low

  * Actualización de parted (para corregir bug al replicar particiones extendidas).
  * Actualización de backharddi-ng-udeb.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Wed, 30 Jun 2010 12:09:27 +0200

backharddi-ng-kernel (0.9) lucid; urgency=low

  * Actualización de partman-base.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Tue, 29 Jun 2010 13:38:01 +0200

backharddi-ng-kernel (0.8) lucid; urgency=low

  * Actualización de backharddi-ng-udeb.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Tue, 29 Jun 2010 12:23:31 +0200

backharddi-ng-kernel (0.7) lucid; urgency=low

  * Actualización de backharddi-ng-udeb.
  * Construcción con mirror de archive.debian.org (distribuciones archivadas).
  * Actualización de debian-installer-utils.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Tue, 29 Jun 2010 08:56:30 +0200

backharddi-ng-kernel (0.6) karmic; urgency=low

  * Actualización de backharddi-ng-udeb.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Sat, 29 May 2010 08:56:02 +0200

backharddi-ng-kernel (0.5) karmic; urgency=low

  * Subida la versión del núcleo a 2.6.32-5-486 de Debian unstable.
  * Actualización de backharddi-ng-udeb.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Fri, 28 May 2010 17:54:02 +0200

backharddi-ng-kernel (0.4) karmic; urgency=low

  * Actualización de backharddi-ng-udeb

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Fri, 21 May 2010 19:13:48 +0200

backharddi-ng-kernel (0.3) karmic; urgency=low

  * Mejoras en el toolchain de construcción del minirt
  * Actualización de partman y dependencias para soporte de ext4
  * Actualización de udpcast
  * Actualización de backharddi-ng-udeb para conectar con el nuevo servicio de red y monitor 

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Fri, 14 May 2010 18:18:43 +0200

backharddi-ng-kernel (0.2) karmic; urgency=low

  * Subida la versión del núcleo a 2.6.32-3-486 de Debian unstable.

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Mon, 03 May 2010 10:18:38 +0200

backharddi-ng-kernel (0.1max1) max; urgency=low

  * Primera versión de backharddi-ng

 -- Pedro Peña Pérez <pedro.pena@open-phoenix.com>  Fri, 18 Sep 2009 22:02:57 +0200

backharddi-kernel (0.12max3) max; urgency=low

  * Fake build in MaX 5.0

 -- Mario Izquierdo (mariodebian) <mariodebian@gmail.com>  Sat, 23 May 2009 17:21:19 +0200

backharddi-kernel (0.12max2) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.17

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Mon,  1 Dec 2008 09:39:06 +0100

backharddi-kernel (0.11max2) unstable; urgency=low

  * Se actualiza cdebconf a versión 0.127backharddi1
  * Se actualiza backharddi-udeb a versión 0.16

 --  <pedro.pena@edmanufacturer.es>  Sat, 22 Nov 2008 15:55:08 +0100

backharddi-kernel (0.10max2) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.15

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Mon, 17 Nov 2008 16:49:17 +0100

backharddi-kernel (0.9max2) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.14

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Fri, 14 Nov 2008 13:39:00 +0100

backharddi-kernel (0.8max2) unstable; urgency=low

  * Se actualizan los paquetes linux-2.6 y linux-kernel-di-i386-2.6 a versión
    2.6.22-6backharddi1edm2 y 1.57backharddi1edm2 respectivamente

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Tue, 11 Nov 2008 16:31:08 +0100

backharddi-kernel (0.8max1) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.13

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Mon, 10 Nov 2008 12:27:29 +0100

backharddi-kernel (0.7max1) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.12

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Thu,  6 Nov 2008 17:30:34 +0100

backharddi-kernel (0.6max1) max; urgency=low

  * Se actualiza rootskel-gtk a versión 1.06backharddi1max1

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Wed, 15 Oct 2008 17:59:36 +0200

backharddi-kernel (0.6) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.11

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Thu, 28 Aug 2008 19:13:39 +0200

backharddi-kernel (0.5) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.10

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Thu, 14 Aug 2008 16:09:25 +0000

backharddi-kernel (0.4) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.9

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Fri,  1 Aug 2008 11:46:48 +0000

backharddi-kernel (0.3) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.8

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Tue, 22 Jul 2008 10:21:28 +0000

backharddi-kernel (0.2) unstable; urgency=low

  * Se actualiza backharddi-udeb a versión 0.3.

 -- Pedro Peña Pérez <pedro.pena@edmanufacturer.es>  Thu, 31 Jan 2008 11:48:03 +0000

backharddi-kernel (0.1) unstable; urgency=low

  * Versión inicial. Núcleo 2.6.22-3-486.


udebs/addons/openvpn/source/debian/openvpn.init.d
#!/bin/sh -e
#
# Original version by Robert Leslie
# <rob@mars.org>, edited by iwj and cs
# Modified for openvpn by Alberto Gonzalez Iniesta <agi@inittab.org>
# Modified for restarting / starting / stopping single tunnels by Richard Mueller <mueller@teamix.net>

test $DEBIAN_SCRIPT_DEBUG && set -v -x

DAEMON=/usr/sbin/openvpn
DESC="virtual private network daemon"
CONFIG_DIR=/etc/openvpn
test -x $DAEMON || exit 0
test -d $CONFIG_DIR || exit 0

# Source defaults file; edit that file to configure this script.
AUTOSTART="all"
STATUSREFRESH=10
if test -e /etc/default/openvpn ; then
  . /etc/default/openvpn
fi

start_vpn () {
    if grep -q '^[	 ]*daemon' $CONFIG_DIR/$NAME.conf ; then
      # daemon already given in config file
      DAEMONARG=
    else
      # need to daemonize
      DAEMONARG="--daemon ovpn-$NAME"
    fi

    if grep -q '^[	 ]*status ' $CONFIG_DIR/$NAME.conf ; then
      # status file already given in config file
      STATUSARG=""
    elif test $STATUSREFRESH -eq 0 ; then
      # default status file disabled in /etc/default/openvpn
      STATUSARG=""
    else
      # prepare default status file
      STATUSARG="--status /var/run/openvpn.$NAME.status $STATUSREFRESH"
    fi

    echo -n " $NAME"
    STATUS="OK"

    $DAEMON --writepid /var/run/openvpn.$NAME.pid \
            $DAEMONARG $STATUSARG --cd $CONFIG_DIR \
            --config $CONFIG_DIR/$NAME.conf < /dev/null || STATUS="FAILED"
    echo -n "($STATUS)"
}
stop_vpn () {
  kill `cat $PIDFILE` || true
  rm $PIDFILE
  rm -f /var/run/openvpn.$NAME.status 2> /dev/null
}

case "$1" in
start)
  echo -n "Starting $DESC:"

  # autostart VPNs
  if test -z "$2" ; then
    # check if automatic startup is disabled by AUTOSTART=none
    if test "x$AUTOSTART" = "xnone" -o -z "$AUTOSTART" ; then
      echo " Autostart disabled."
      exit 0
    fi
    if test -z "$AUTOSTART" -o "x$AUTOSTART" = "xall" ; then
      # all VPNs shall be started automatically
      for CONFIG in `cd $CONFIG_DIR; ls *.conf 2> /dev/null`; do
        NAME=${CONFIG%%.conf}
        start_vpn
      done
    else
      # start only specified VPNs
      for NAME in $AUTOSTART ; do
        if test -e $CONFIG_DIR/$NAME.conf ; then
          start_vpn
        else
          echo -n " (failure: No such VPN: $NAME)"
        fi
      done
    fi
  #start VPNs from command line
  else
    while shift ; do
      [ -z "$1" ] && break
      if test -e $CONFIG_DIR/$1.conf ; then
        NAME=$1
        start_vpn
      else
        echo -n " (failure: No such VPN: $1)"
      fi
    done
  fi
  echo "."

  ;;
stop)
  echo -n "Stopping $DESC:"

  if test -z "$2" ; then
    for PIDFILE in `ls /var/run/openvpn.*.pid 2> /dev/null`; do
      NAME=`echo $PIDFILE | cut -c18-`
      NAME=${NAME%%.pid}
      stop_vpn
      echo -n " $NAME"
    done
  else
    while shift ; do
      [ -z "$1" ] && break
      if test -e /var/run/openvpn.$1.pid ; then
        PIDFILE=`ls /var/run/openvpn.$1.pid 2> /dev/null`
        NAME=`echo $PIDFILE | cut -c18-`
        NAME=${NAME%%.pid}
        stop_vpn
        echo -n " $NAME"
      else
        echo -n " (failure: No such VPN is running: $1)"
      fi
    done
  fi
  echo "."
  ;;
# We only 'reload' for running VPNs. New ones will only start with 'start' or 'restart'.
reload|force-reload)
  echo -n "Reloading $DESC:"
  for PIDFILE in `ls /var/run/openvpn.*.pid 2> /dev/null`; do
    NAME=`echo $PIDFILE | cut -c18-`
    NAME=${NAME%%.pid}
# If openvpn if running under a different user than root we'll need to restart
    if egrep '^( |\t)*user' $CONFIG_DIR/$NAME.conf > /dev/null 2>&1 ; then
      stop_vpn
      sleep 1
      start_vpn
      echo -n "(restarted)"
    else
      kill -HUP `cat $PIDFILE` || true
    echo -n " $NAME"
    fi
  done
  echo "."
  ;;

restart)
  shift
  $0 stop ${@}
  sleep 1
  $0 start ${@}
  ;;
cond-restart)
  echo -n "Restarting $DESC:"
  for PIDFILE in `ls /var/run/openvpn.*.pid 2> /dev/null`; do
    NAME=`echo $PIDFILE | cut -c18-`
    NAME=${NAME%%.pid}
    stop_vpn
    sleep 1
    start_vpn
  done
  echo "."
  ;;
*)
  echo "Usage: $0 {start|stop|reload|restart|force-reload|cond-restart}" >&2
  exit 1
  ;;
esac

exit 0

# vim:set ai sts=2 sw=2 tw=0:
